src/fragment/factory.js

```diff
@@ -41,6 +41,15 @@ export default function FragmentFactory (vm, el) {
   this.linker = linker
 }
 
+function hasDirectives (el) {
```

Comment:
this could be simpler, since `v-for` and `v-if` are the only ones that can work on `<template>`, and `v-for` has higher priority than `v-if`, it seems `v-if` is the only thing that we need to check.